wp-cli/class-vip-go-async-scheduler-command.php

```diff
@@ -0,0 +1,183 @@
+			wp_schedule_single_event( $timestamp, self::SCHEDULE_EVENT_KEY, [ $command ] );
+			wp_cache_set( $cache_key, $timestamp, self::SCHEDULE_TIMESTAMP_KEY );
+		} else {
+			WP_CLI::error( 'This command is already scheduled or running.' );
+		}
+	}
+
+	/**
+	 * Check the status of the command
+	 *
+	 * @subcommand check-status
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--cmd]
+	 * : command to run on schedule, without `wp`
+	 */
+	public function check( $args, $assoc_args ) {
+		self::validate_args( $args, $assoc_args );
+
+		$command   = self::normalize_command( $assoc_args['cmd'] );
+		$cache_key = md5( $command );
+
+		// This is not guaranteed to exist, but worth a shot.
+		$timestamp = wp_cache_get( $cache_key, self::SCHEDULE_TIMESTAMP_KEY );
+
+		$events = self::get_commands( $command );
+
+		if ( ! $events ) {
+			WP_CLI::error( 'No events found' );
+		}
+
+		// phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition.Found,Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure
+		if ( $timestamp ) {
+			$event_idx = array_search( (string) $timestamp, array_column( $events, 'timestamp' ) );
+
+			if ( null !== $event_idx && isset( $events[ $event_idx ] ) ) {
+				WP_CLI::success( sprintf( 'Found event, status: %s; created: %s; last modified: %s', $events[ $event_idx ]->status, $events[ $event_idx ]->created, $events[ $event_idx ]->last_modified ) );
+				exit;
+			}
+		} else {
+			WP_CLI::warning( "Exact match couldn't be found, but there are commands with the same arguments:" );
+			foreach ( $events as $event ) {
+				WP_CLI::line( sprintf( 'Status: %s; created: %s; last modified: %s', $event->status, $event->created, $event->last_modified ) );
+			}
+		}
+	}
+
+	/**
+	 * This is the runner itself.
+	 *
+	 * @param string $command
+	 * @return void
+	 */
+	public static function runner( $command ) {
+		$cache_key = md5( $command );
+
+		$result = WP_CLI::runcommand(
+			$command,
+			[
+				// Grab an object that includes STDERR, STDOUT and exit code
+				'return'     => 'all',
+				// Do not exit on error
+				'exit_error' => false,
+			]
+		);
+
+		wp_cache_delete( $cache_key, self::SCHEDULE_TIMESTAMP_KEY );
+
+		if ( $result->stderr ) {
+			WP_CLI::warning( $result->stderr );
+		}
+
+		if ( 0 !== $result->return_code ) {
+			WP_CLI::warning( sprintf( 'The scheduled command `%s` has non-zero exit code', $command, $result->return_code ) );
+		}
+	}
+
+	/**
+	 * Grab any events we have stored that match the command by hook and the argument (the command to be ran).
+	 *
+	 * @param string $command
+	 * @return array 
+	 */
+	public static function get_commands( string $command ): array {
+		global $wpdb;
+
+		$command      = self::normalize_command( $command );
+		$events_store = Events_Store::instance();
+
+		// phpcs:ignore 
+		$events = $wpdb->get_results( $wpdb->prepare( "SELECT * FROM {$events_store->get_table_name()} WHERE action = %s AND args = %s LIMIT 10", self::SCHEDULE_EVENT_KEY, maybe_serialize( [ $command ] ) ) );
```

Comment:
Would it make sense to move this to the Events_Store class? Just to keep the single responsibility. I know there are already a few methods there to get jobs, and none of them fits here, but even if we have to add another one, at least it might get reused (or refactored) in the future.

Reply:
Just wanted to limit the scope of changes and, since it's more of a temporary thing, introducing potential cruft in the Cron Control didn't seem worth it.

Reply:
If it's something temporary, then it certainly makes sense :)